pkg/scheduler/scheduler.go

```diff
@@ -131,6 +131,14 @@ type ScaleCacheConfig struct {
 	RefreshPeriod time.Duration `json:"refreshPeriod"`
 }
 
+func (c ScaleCacheConfig) withDefaults() ScaleCacheConfig {
+	n := ScaleCacheConfig{}
+	if c.RefreshPeriod == 0 {
+		n.RefreshPeriod = 5 * time.Minute
+	}
+	return n
```

Comment:
Wouldn't this return a `ScaleCacheConfig` without the refresh period set if `c.RefreshPeriod != 0`?

I think we should add a `else return c` or something similar

Reply:
Fixed